src/common/src/config.rs

```diff
@@ -240,6 +240,10 @@ pub struct MetaConfig {
 
     #[serde(default, flatten)]
     pub unrecognized: Unrecognized<Self>,
+
+    /// Whether config object storage buckey lifecycle to purge stale data.
+    #[serde(default)]
+    pub do_not_config_object_storage_lifecycle: bool,
```

Comment:
nits: I think using a config named `configure_s3_bucket_lifecycle` is more straightfoward rather than using a config to disable it.

Reply:
Can we also put an entry in [OverrideConfigOpts](https://github.com/risingwavelabs/risingwave/blob/094dade370a1cc522d22c23adadfc6b0193e7091/src/meta/src/lib.rs#L127) so that cloud can easily override this config with env variable?

Reply:
I think the default behavior is to config lifecycle, while the default value of bool is `false`, so it is named.